### PR TITLE
Changed EdgeCost implementation to include calculated time and road type

### PR DESCRIPTION
**Pull request by aroni11**

### CODE REVIEW

File: src/algorithms/models/edgecost.ts [hunk 1]
```diff
@@ -7,18 +9,111 @@ class EdgeCost {
...
+
   static combine(...ecs: EdgeCost[]): EdgeCost {
     const res = new EdgeCost();
     for (const ec of ecs) {
       res.distance += ec.distance;
+      res.time += ec.time;
+      res.road_cost += ec.distance * ec.road_cost;
     }
+
+    res.road_cost /= res.distance;
```
Comment (HonzaKopecky) [inline src/algorithms/models/edgecost.ts:21]: Cool, my original idea was to have something like sum variables but this is actually better!

File: src/algorithms/models/edgecost.ts [hunk 2]
```diff
@@ -7,18 +9,111 @@ class EdgeCost {
...
+
     return res;
   }
 
   /**
    * Real distance between two vertices in meters
+   * Real time between two vertices in minutes
+   * Type of road between two vertices as a float value
+   */
```
Comment (HonzaKopecky) [inline src/algorithms/models/edgecost.ts]: The comments should be separated. One comment for every variable.

File: src/algorithms/models/edgecost.ts [hunk 3]
```diff
@@ -7,18 +9,111 @@ class EdgeCost {
...
+      return;
+    }
+
+    const lngSq = Math.pow((v1.lng - v2.lng), 2);
+    const latSq = Math.pow((v1.lat - v2.lat), 2);
+    this.distance = Math.sqrt(latSq + lngSq);
+
+    let max_speed: number;
+
+    switch (road_type) {
```
Comment (HonzaKopecky) [inline src/algorithms/models/edgecost.ts:57]: Let's move the switch to a separate private method. So the constructor is not that looong.

**Review by HonzaKopecky — CHANGES_REQUESTED** [review]
My comments added, have a look please ;)

File: src/algorithms/models/edgecost.ts (hunk 2 above)
Comment (aroni11) [reply src/algorithms/models/edgecost.ts]: Agreed. Done.

File: src/algorithms/models/edgecost.ts (hunk 1 above)
Comment (aroni11) [reply src/algorithms/models/edgecost.ts:21]: the-most-interesting-man-in-the-world

File: src/algorithms/models/edgecost.ts (hunk 3 above)
Comment (aroni11) [reply src/algorithms/models/edgecost.ts:57]: Done

**Pull request merged**